src/devices/MatrixKeyboard/MatrixKeyboard.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Threading;
+
+namespace Iot.Device.MatrixKeyboard
+{
+    /// <summary>
+    /// GPIO Matrix Keyboard Driver
+    /// </summary>
+    public class MatrixKeyboard : IDisposable
+    {
+        /// <summary>
+        /// Get row pins
+        /// </summary>
+        public IEnumerable<int> RowPins => _rowPins;
+
+        /// <summary>
+        /// Get column pins
+        /// </summary>
+        public IEnumerable<int> ColumnPins => _columnPins;
+
+        /// <summary>
+        /// Get or set scanning frequency in Hertz
+        /// </summary>
+        public double ScanFrequency
+        {
+            get => _scanFrequency;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), "Scanning frequency must be positive");
+                }
+                _scanFrequency = value;
+                _scanInterval = CalcScanInterval(_scanFrequency);
+            }
+        }
+
+        /// <summary>
+        /// Get whether this keyboard is scanning
+        /// </summary>
+        public bool IsScanning { get; private set; }
+
+        private readonly int[] _rowPins;
+        private readonly int[] _columnPins;
+        private int _currentRow;
+        private readonly PinValue[] _buttonValues;
+        private double _scanFrequency;
+
+        /// <summary>
+        /// Initialize keyboard
+        /// </summary>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="rowPins">Row pins</param>
+        /// <param name="colPins">Column pins</param>
+        /// <param name="scanFreq">Scanning frequency</param>
+        public MatrixKeyboard(IGpioController gpioController, IEnumerable<int> rowPins, IEnumerable<int> colPins, double scanFreq = 50)
```

Comment:
scanFreq should have a unit in the name

Reply:
also, would it make sense to change it to interval?

Reply:
you mean `int interval` instead of `double frequency`?